hazelcast/src/main/java/com/hazelcast/map/impl/proxy/MapProxySupport.java

```diff
@@ -1391,14 +1395,19 @@ protected <T extends Result> T executeQueryInternal(Predicate predicate, Aggrega
 
         if (predicate instanceof PartitionPredicate) {
             PartitionPredicate partitionPredicate = (PartitionPredicate) predicate;
-            Data key = toData(partitionPredicate.getPartitionKey());
-            int partitionId = partitionService.getPartitionId(key);
-            if (target.mode() == TargetMode.LOCAL_NODE && !partitionService.isPartitionOwner(partitionId)
-                    || target.mode() == TargetMode.PARTITION_OWNER && !target.partitions().contains(partitionId)
-            ) {
+            PartitionIdSet partitionIds = new PartitionIdSet(partitionService.getPartitionCount());
+            boolean allFalse = true;
+            for (Object o : partitionPredicate.getPartitionKeys()) {
+                Data key = toData(o);
+                int partitionId = partitionService.getPartitionId(key);
+                partitionIds.add(partitionId);
+                allFalse &= target.mode() == TargetMode.LOCAL_NODE && !partitionService.isPartitionOwner(partitionId)
```

Comment:
Is it ok to create a short-circuit here? Seems after first `false`, loop can finish, instead of iterating whole given partition key set

Reply:
`allFalse` isn't used as the loop control variable.

This code is inside the `for (Object o : partitionPredicate.getPartitionKeys())` iteration, so it will be evaluated for every object in `getPartitionKeys`.

After the iterator has finished, this variable will be true if and only if all of the predicates would evaluate to false (either the target mode is local and the partition id isn't the owner, or the target mode is partition owner but the target doesn't contain the partition id)

Reply:
I didn't get why can't we do that. Were the comments for an old version of the implementation in this PR? I feel like we can short-circuit the loop with the current state